spec/reek/configuration/configuration_file_finder_spec.rb

```diff
@@ -1,34 +1,58 @@
-require 'spec_helper'
-require 'reek/cli/application'
+# encoding: UTF-8
 
-include Reek::Cli
-include Reek::Configuration
+require 'pathname'
+require 'tmpdir'
+require 'spec_helper'
 
 describe ConfigurationFileFinder do
   describe '.find' do
-    let(:sample_config_path) { Pathname.new('spec/samples/simple_configuration.reek') }
-    let(:config_path_same_level) { Pathname.new('spec/samples/simple_configuration.reek') }
-    let(:options_with_config_file) { double(config_file: sample_config_path) }
-    let(:options_without_config_file) { double(config_file: nil) }
+    it 'returns the config_file if it’s set' do
+      config_file = double
+      found = ConfigurationFileFinder.find(double(config_file: config_file))
+      expect(found).to eq(config_file)
+    end
 
-    it 'should return the config file we passed as cli option if given' do
-      expect(ConfigurationFileFinder.find(options_with_config_file)).to eq(sample_config_path)
+    it 'returns the file in current dir if config_file is nil' do
+      options = double(config_file: nil)
+      current = Pathname.new('spec/samples')
+      found = ConfigurationFileFinder.find(options, current: current)
+      expect(found).to eq(Pathname.new('spec/samples/exceptions.reek'))
     end
 
-    it 'should return the first configuration file it can find in the current directory' do
-      allow(ConfigurationFileFinder).to receive(:detect_or_traverse_up).
-        and_return config_path_same_level
+    it 'returns the file in current dir if options is nil' do
+      current = Pathname.new('spec/samples')
+      found = ConfigurationFileFinder.find(nil, current: current)
```

Comment:
Can we move the "options" hash into the second hash? It's really confusing when a methods first parameter is nil. It always makes me think "if I need to pass nil in here just to make it work, maybe this parameter shouldn't be so prominent?".